PwaUriHandler/explainer.md

```diff
@@ -110,28 +112,30 @@ These are the fields in each URI handler object:
+| `excludePaths` | Optional       | Array of disallowed paths relative to `host`     | `[]`                              |
+
+`host` must match valid hosts under [RFC 3986](https://tools.ietf.org/html/rfc3986). `path` and `excludePaths` must match valid paths. If a scheme is provided in `host`, it must be __https://__ and not __http://__. Otherwise, __https://__ will be assumed.
 
-In this scheme, a URI matches a URI handler if it matches the `base`, at least one of values in `paths` if there are any, does not match any of the values in `excludePaths`, and is [verifiably associated](#pwa-to-site-association) with the PWA.
+In this scheme, a URI matches a URI handler if it matches the `host`, at least one of values in `paths` if there are any, does not match any of the values in `excludePaths`, and is [verifiably associated](#pwa-to-site-association) with the PWA.
 
-Requested URIs do not have to be within the requesting PWA's scope. In this scheme, any URI can be registered as part of the URI handling request. The `base` field is necessary because URIs from different domains can be requested. Not restricting URIs to the same scope or domain as the requesting PWA gives the developer freedom to use multiple domain names for the same content and handle them with the same PWA. See [this section](#pwa-to-site-association) for how cross-domain requests are validated. The `base` field can start with a `%*.` prefix to indicate the inclusion of subdomains.
+Requested URIs do not have to be within the requesting PWA's scope. In this scheme, any URI can be registered as part of the URI handling request. The `host` field is necessary because URIs from different domains can be requested. Not restricting URIs to the same scope or domain as the requesting PWA gives the developer freedom to use multiple domain names for the same content and handle them with the same PWA. See [this section](#pwa-to-site-association) for how cross-domain requests are validated. The `host` field can start with a `*.` prefix to indicate the inclusion of subdomains.
 
 (Implementation note: URI handling requests are registered with either the browser or the OS when a PWA is being installed. At this point, the browser should validate the requests. If necessary, the PWA install can be failed.)
 
 #### Wildcard Matching
 
-Wildcard characters can be used in the values of all three fields: `base`, `paths`, and `excludePaths`. The wildcard `*` matches zero or more characters. The wildcard `?` matches exactly one character.
+Wildcard characters can be used in the values of all three fields: `host`, `paths`, and `excludePaths`. The wildcard character `*` matches zero or more characters. Wildcard can only be used as a prefix for `host`. Wildcard can only be used as a postfix for `paths` and `excludePaths`. In-fix wildcards are disallowed to protect against poor performance during string matching.
```

Comment:
What happens if someone does violates these rules?

Reply:
Those handler objects should not be registered. Edited.